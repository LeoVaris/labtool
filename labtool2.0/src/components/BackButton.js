import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { Button, Icon } from 'semantic-ui-react'

const BackButton = props => (
  <Link to={props.to}>
    <Button compact disabled={!props.enabled} onClick={props.cleanup || (() => {})}>
      <Icon name="angle left" />
      <span>{props.text}</span>
    </Button>
  </Link>
)

BackButton.propTypes = {
  to: PropTypes.string,
  enabled: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
  cleanup: PropTypes.func
}

const presets = {
  modifyCIPage: {
    to: state => (state.selectedInstance.ohid ? `/labtool/ModifyCourseInstancePage/${state.selectedInstance.ohid}` : `/labtool/courses`),
    text: 'Back to course editing'
  },
  coursePage: {
    to: state => (state.selectedInstance.ohid ? `/labtool/courses/${state.selectedInstance.ohid}` : `/labtool/courses`),
    text: 'Back to course page'
  }
}

const mapStateToProps = (state, ownProps) => {
  let to
  try {
    to = presets[ownProps.preset].to(state)
  } catch (e) {
    to = ownProps.to
  }
  let text
  if (presets[ownProps.preset]) {
    text = presets[ownProps.preset].text
  } else {
    text = ownProps.text || 'Back'
  }
  return {
    to,
    text,
    enabled: to !== undefined
  }
}

export default connect(
  mapStateToProps,
  null
)(BackButton)
