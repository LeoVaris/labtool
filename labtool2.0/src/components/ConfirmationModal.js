import React, { useState } from 'react'
import { Modal, Button, Icon } from 'semantic-ui-react'
import PropTypes from 'prop-types'

//used in ModifyCourseInstanceCodeReviews
export const ConfirmationModal = props => {
  const [open, setOpen] = useState(false)

  const toggleOpen = () => setOpen(true)
  const toggleClose = () => setOpen(false)

  return (
    <Modal size="tiny" open={open} trigger={<Icon id="tag" onClick={toggleOpen} name="window close" size="large" color="red" style={{ float: 'right' }} />}>
      <Modal.Content>
        <Modal.Description>
          <p>Do you wish to remove the following code review:</p>
          <p>
            {props.data.User.firsts} {props.data.User.lastname} reviewing {props.getCurrentReviewer(props.selectedDropdown, props.data.id)}
          </p>
        </Modal.Description>
      </Modal.Content>
      <Modal.Actions>
        <Button negative icon="close" labelPosition="right" color="red" content="No" onClick={toggleClose} />
        <Button positive icon="checkmark" labelPosition="right" content="Yes" onClick={props.removeOne(props.data.id)} />
      </Modal.Actions>
    </Modal>
  )
}
ConfirmationModal.propTypes = {
  data: PropTypes.object.isRequired,
  selectedDropdown: PropTypes.number.isRequired,
  getCurrentReviewer: PropTypes.func.isRequired,
  removeOne: PropTypes.func.isRequired
}
export default ConfirmationModal
