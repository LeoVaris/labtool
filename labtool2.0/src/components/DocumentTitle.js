import React from 'react'
import { Helmet } from 'react-helmet'

export default ({ title }) => (
  <Helmet titleTemplate={'%s | Labtool 2.0'} defaultTitle={'Labtool 2.0'}>
    <title>{title}</title>
  </Helmet>
)
