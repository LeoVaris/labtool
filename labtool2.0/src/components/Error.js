import React from 'react'
import PropTypes from 'prop-types'
import { Message, List } from 'semantic-ui-react'

export const LabtoolError = ({ header, errors }) => (
  <Message
    error
    icon="warning sign"
    header={header || 'Error'}
    content={
      <List bulleted>
        {errors.map(error => (
          <List.Item key={error}>{error}</List.Item>
        ))}
      </List>
    }
  />
)
LabtoolError.propTypes = {
  header: PropTypes.string,
  errors: PropTypes.array
}

export default LabtoolError
