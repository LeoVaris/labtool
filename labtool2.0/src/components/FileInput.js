import React from 'react'
import PropTypes from 'prop-types'
import { Button, Modal, Form, Progress, Message } from 'semantic-ui-react'

export default class FileInput extends React.Component {
  state = { open: false, uploading: false, uploadProgress: null, uploadError: null }

  constructor(props) {
    super(props)

    this.fileReader = new FileReader()
    this.fileReader.onloadstart = () => this.setState({ uploading: true, uploadProgress: 0 })
    this.fileReader.onprogress = progress => {
      if (progress.lengthComputable) {
        this.setState({ uploadProgress: 100 * (progress.loaded / progress.total) })
      }
    }
    this.fileReader.onerror = () => this.setState({ uploadError: this.fileReader.error.message, uploadProgress: null })
    this.fileReader.onload = () => {
      this.props.onFileUploaded(this.fileReader.result)
      this.close()
    }
  }

  open = () => this.setState({ open: true })
  close = () => {
    this.fileReader.abort()
    this.setState({ open: false, uploading: false, uploadProgress: null, uploadError: null })
  }

  handleFile = event => {
    if (event.target.files && event.target.files[0]) {
      this.fileReader.readAsArrayBuffer(event.target.files[0])
    }
  }

  render() {
    const { allowedFileTypes, style, ...props } = this.props
    const { open, uploading, uploadProgress, uploadError } = this.state

    return (
      <React.Fragment>
        <Button disabled={open} onClick={this.open} style={style} {...props}>
          Upload
        </Button>

        <Modal onClose={this.close} open={open}>
          <Modal.Header>Select file</Modal.Header>
          <Modal.Description style={{ padding: 15 }}>
            {!uploading && (
              <Form>
                <Form.Input type="file" name="file" accept={allowedFileTypes && allowedFileTypes.join()} onChange={this.handleFile} onClick={e => (e.target.value = null)} />
              </Form>
            )}
            {uploading && uploadProgress !== null && <Progress percent={uploadProgress} indicating />}
            {uploading &&
              uploadError && (
                <Message error>
                  <p>{this.state.uploadError}</p>
                </Message>
              )}
          </Modal.Description>
          <Modal.Actions>
            <Button content="Close" negative onClick={this.close} />
          </Modal.Actions>
        </Modal>
      </React.Fragment>
    )
  }
}

FileInput.propTypes = {
  onFileUploaded: PropTypes.func.isRequired,
  allowedFileTypes: PropTypes.arrayOf(PropTypes.string),
  style: PropTypes.object
}
