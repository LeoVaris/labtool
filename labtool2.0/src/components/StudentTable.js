import React from 'react'
import PropTypes from 'prop-types'
import { Button, Icon, Table, Popup, Dropdown, Label, Checkbox } from 'semantic-ui-react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import HorizontalScrollable from './HorizontalScrollable'
import { getAllTags, tagStudent, unTagStudent } from '../services/tags'
import { associateTeacherToStudent } from '../services/assistant'
import {
  showAssistantDropdown,
  showTagDropdown,
  filterByTag,
  filterByAssistant,
  selectTeacher,
  selectTag,
  selectStudent,
  unselectStudent,
  selectAllStudents,
  unselectAllStudents
} from '../reducers/coursePageLogicReducer'
import { createDropdownTeachers, createDropdownTags } from '../util/dropdown'

const { Fragment } = React

export const StudentTable = props => {
  const updateTeacher = id => async e => {
    try {
      e.preventDefault()
      let teacherId = props.coursePageLogic.selectedTeacher
      if (teacherId === '-') {
        // unassign
        teacherId = null
      }

      const data = {
        studentInstanceId: id,
        teacherInstanceId: teacherId
      }
      await props.associateTeacherToStudent(data)
    } catch (error) {
      console.error(error)
    }
  }

  const handleSelectCheck = id => (e, data) => {
    const { checked } = data
    if (checked) {
      props.selectStudent(id)
    } else {
      props.unselectStudent(id)
    }
  }

  const handleSelectAll = filteredData => (e, data) => {
    const { checked } = data
    // we use filteredData so that we only select or deselect
    // only the filtered entries. sneaky!
    const ids = filteredData.map(data => data.id)
    if (checked) {
      props.selectAllStudents(ids)
    } else {
      props.unselectAllStudents(ids)
    }
  }

  const changeSelectedTeacher = () => {
    return (e, data) => {
      const { value } = data
      props.selectTeacher(value)
    }
  }

  const changeSelectedTag = () => {
    return (e, data) => {
      const { value } = data
      props.selectTag(value)
    }
  }

  const changeHiddenAssistantDropdown = id => {
    return () => {
      props.showAssistantDropdown(props.coursePageLogic.showAssistantDropdown === id ? '' : id)
    }
  }

  const changeHiddenTagDropdown = id => {
    return () => {
      props.showTagDropdown(props.coursePageLogic.showTagDropdown === id ? '' : id)
    }
  }

  const addTag = id => async e => {
    try {
      e.preventDefault()
      const data = {
        studentId: id,
        tagId: props.coursePageLogic.selectedTag
      }
      await props.tagStudent(data)
    } catch (error) {
      console.error(error)
    }
  }

  const removeTag = id => async e => {
    try {
      e.preventDefault()
      const data = {
        studentId: id,
        tagId: props.coursePageLogic.selectedTag
      }
      await props.unTagStudent(data)
    } catch (error) {
      console.error(error)
    }
  }

  const changeFilterAssistant = () => {
    return (e, data) => {
      const { value } = data
      props.filterByAssistant(value)
    }
  }

  const addFilterTag = tag => {
    return () => {
      props.filterByTag(tag)
    }
  }

  const hasFilteringTags = (studentTagsData, filteringTags) => {
    let studentInstanceTagIds = studentTagsData.map(tag => tag.id)
    let filteringTagIds = filteringTags.map(tag => tag.id)
    let hasRequiredTags = true
    for (let i = 0; i < filteringTagIds.length; i++) {
      if (!studentInstanceTagIds.includes(filteringTagIds[i])) {
        hasRequiredTags = false
      }
    }
    return hasRequiredTags
  }

  const shouldHideInstructor = studentInstances => studentInstances.every(studentInstance => studentInstance.teacherInstanceId === null)

  const createHeadersTeacher = () => {
    const headers = []
    let i = 0
    for (; i < props.selectedInstance.weekAmount; i++) {
      headers.push(
        <Table.HeaderCell key={i}>
          Week
          <br />
          {i + 1}{' '}
        </Table.HeaderCell>
      )
    }
    for (var ii = 1; ii <= props.selectedInstance.amountOfCodeReviews; ii++) {
      headers.push(
        <Table.HeaderCell key={i + ii}>
          Code
          <br />
          Review
          <br />
          {ii}{' '}
        </Table.HeaderCell>
      )
    }
    if (props.selectedInstance.finalReview) {
      headers.push(
        <Table.HeaderCell key={i + ii + 1}>
          Final
          <br />
          Review{' '}
        </Table.HeaderCell>
      )
    }
    return headers
  }

  const createIndents = (weeks, codeReviews, siId) => {
    const cr =
      codeReviews &&
      codeReviews.reduce((a, b) => {
        return { ...a, [b.reviewNumber]: b.points }
      }, {})
    const indents = []
    let i = 0
    let finalPoints = undefined
    for (; i < props.selectedInstance.weekAmount; i++) {
      let pushattava = (
        <Table.Cell key={'week' + i}>
          <p>-</p>
        </Table.Cell>
      )

      for (var j = 0; j < weeks.length; j++) {
        if (i + 1 === weeks[j].weekNumber) {
          pushattava = (
            <Table.Cell key={i + ':' + j}>
              <p>{weeks[j].points}</p>
            </Table.Cell>
          )
        } else if (weeks[j].weekNumber === props.selectedInstance.weekAmount + 1) {
          finalPoints = weeks[j].points
        }
      }
      indents.push(pushattava)
    }

    let ii = 0
    const { amountOfCodeReviews } = props.selectedInstance
    if (amountOfCodeReviews) {
      for (let index = 1; index <= amountOfCodeReviews; index++) {
        indents.push(<Table.Cell key={siId + index}>{cr[index] || cr[index] === 0 ? <p className="codeReviewPoints">{cr[index]}</p> : <p>-</p>}</Table.Cell>)
      }
    }

    if (props.selectedInstance.finalReview) {
      let finalReviewPointsCell = (
        <Table.Cell key={i + ii + 1}>
          <p>{finalPoints === undefined ? '-' : finalPoints}</p>
        </Table.Cell>
      )
      indents.push(finalReviewPointsCell)
    }

    return indents
  }

  const createStudentTableRow = (showColumn, data, rowClassName, dropDownTags, dropDownTeachers) => (
    <Table.Row key={data.id} className={rowClassName}>
      {/* Select Check Box */}
      {showColumn('select') && (
        <Table.Cell key="select">
          <Checkbox id={'select' + data.id} checked={props.coursePageLogic.selectedStudents[data.id] || false} onChange={handleSelectCheck(data.id)} />
        </Table.Cell>
      )}

      {/* Student */}
      <Table.Cell key="studentinfo">
        {data.User.firsts} {data.User.lastname} ({data.User.studentNumber})
      </Table.Cell>

      {/* Project Info */}
      <Table.Cell key="projectinfo">
        <span>
          {data.projectName}
          <br />
          <a href={data.github} target="_blank" rel="noopener noreferrer">
            {data.github}
          </a>
          {data.Tags.map(tag => (
            <div key={data.id + ':' + tag.id}>
              <Button compact floated="left" className={`mini ui ${tag.color} button`} onClick={addFilterTag(tag)}>
                {tag.name}
              </Button>
            </div>
          ))}
          {props.allowModify && (
            <Popup trigger={<Icon id="tag" onClick={changeHiddenTagDropdown(data.id)} name="pencil" color="green" style={{ float: 'right', fontSize: '1.25em' }} />} content="Add or remove tag" />
          )}
        </span>
        {props.allowModify && (
          <div>
            {props.coursePageLogic.showTagDropdown === data.id ? (
              <div>
                <Dropdown id="tagDropdown" style={{ float: 'left' }} options={dropDownTags} onChange={changeSelectedTag()} placeholder="Choose tag" fluid selection />
                <br />
                <div className="two ui buttons" style={{ float: 'left' }}>
                  <button className="ui icon positive button" onClick={addTag(data.id)} size="mini">
                    <i className="plus icon" />
                  </button>
                  <div className="or" />
                  <button className="ui icon button" onClick={removeTag(data.id)} size="mini">
                    <i className="trash icon" />
                  </button>
                </div>
              </div>
            ) : (
              <div />
            )}
          </div>
        )}
      </Table.Cell>

      {showColumn('points') && (
        <Fragment>
          {/* Week #, Code Review # */}
          {createIndents(data.weeks, data.codeReviews, data.id)}

          {/* Sum */}
          <Table.Cell key="pointssum">
            {(data.weeks.map(week => week.points).reduce((a, b) => a + b, 0) + data.codeReviews.map(cr => cr.points).reduce((a, b) => a + b, 0)).toFixed(2).replace(/[.,]00$/, '')}
          </Table.Cell>
        </Fragment>
      )}

      {/* Instructor */}
      {!shouldHideInstructor(props.studentInstances) && (
        <Table.Cell key="instructor">
          {data.teacherInstanceId && props.selectedInstance.teacherInstances ? (
            props.selectedInstance.teacherInstances
              .filter(teacher => teacher.id === data.teacherInstanceId)
              .map(teacher => (
                <span key={data.id + ':' + teacher.id}>
                  {teacher.firsts} {teacher.lastname}
                </span>
              ))
          ) : (
            <span>not assigned</span>
          )}
          {props.allowModify && (
            <Fragment>
              <Popup
                trigger={<Button circular onClick={changeHiddenAssistantDropdown(data.id)} size="small" icon={{ name: 'pencil' }} style={{ margin: '0.25em', float: 'right' }} />}
                content="Assign instructor"
              />
              {props.coursePageLogic.showAssistantDropdown === data.id ? (
                <div>
                  <Dropdown id="assistantDropdown" options={dropDownTeachers} onChange={changeSelectedTeacher()} placeholder="Select teacher" fluid selection />
                  <Button onClick={updateTeacher(data.id, data.teacherInstanceId)} size="small">
                    Change instructor
                  </Button>
                </div>
              ) : (
                <div />
              )}
            </Fragment>
          )}
        </Table.Cell>
      )}

      {showColumn('review') && (
        <Fragment>
          {/* Review */}
          <Table.Cell key="reviewbutton" textAlign="right">
            <Link to={`/labtool/browsereviews/${props.selectedInstance.ohid}/${data.id}`}>
              <Popup trigger={<Button circular size="tiny" icon={{ name: 'star', size: 'large', color: 'orange' }} />} content="Review student" />
            </Link>
          </Table.Cell>
        </Fragment>
      )}
    </Table.Row>
  )

  const { columns, rowClassName, disableDefaultFilter } = props

  const showColumn = column => columns.indexOf(column) >= 0

  let dropDownTeachers = []
  dropDownTeachers = createDropdownTeachers(props.selectedInstance.teacherInstances, dropDownTeachers)
  let dropDownFilterTeachers = [
    {
      key: 0,
      text: 'no filter',
      value: 0
    }
  ]
  dropDownFilterTeachers = createDropdownTeachers(props.selectedInstance.teacherInstances, dropDownFilterTeachers)

  let dropDownTags = []
  dropDownTags = createDropdownTags(props.tags.tags, dropDownTags)

  const filteredData = (props.studentInstances || [])
    // remove students when filtering assistants and it doesn't match
    .filter(
      data =>
        disableDefaultFilter ||
        props.coursePageLogic.filterByAssistant === 0 ||
        props.coursePageLogic.filterByAssistant === data.teacherInstanceId ||
        (props.coursePageLogic.filterByAssistant === '-' && data.teacherInstanceId === null) // unassign = -
    )
    // remove students when filtering tags and they don't match
    .filter(data => disableDefaultFilter || props.coursePageLogic.filterByTag.length === 0 || hasFilteringTags(data.Tags, props.coursePageLogic.filterByTag))

  // all students currently visible selected?
  const allSelected = filteredData.map(data => data.id).every(id => props.coursePageLogic.selectedStudents[id])

  // calculate the length of the longest text in a drop down
  const getBiggestWidthInDropdown = dropdownList => {
    if (dropdownList.length === 0) {
      return 3
    }
    const lengths = dropdownList.map(dp => dp.text.length)
    return lengths.reduce((longest, comp) => (longest > comp ? longest : comp), lengths[0])
  }

  return (
    <Fragment>
      <div style={{ textAlign: 'left' }}>
        <span>Filter by instructor </span>
        <Dropdown
          scrolling
          options={dropDownFilterTeachers}
          onChange={changeFilterAssistant()}
          placeholder="Select Teacher"
          defaultValue={props.coursePageLogic.filterByAssistant}
          selection
          style={{ width: `${getBiggestWidthInDropdown(dropDownFilterTeachers)}em` }}
        />
        <span> Tag filters: </span>

        {props.coursePageLogic.filterByTag.length === 0 ? (
          <span>
            <Label>none</Label>
          </span>
        ) : (
          <span>
            {props.coursePageLogic.filterByTag.map(tag => (
              <span key={tag.id}>
                <Button compact className={`mini ui ${tag.color} button`} onClick={addFilterTag(tag)}>
                  {tag.name}
                </Button>
              </span>
            ))}
          </span>
        )}
      </div>

      <HorizontalScrollable>
        <Table celled compact unstackable singleLine style={{ overflowX: 'visible' }}>
          <Table.Header>
            <Table.Row>
              {showColumn('select') && (
                <Table.HeaderCell key={-2}>
                  <Checkbox id="selectAll" checked={allSelected} onChange={handleSelectAll(filteredData)} />
                </Table.HeaderCell>
              )}
              <Table.HeaderCell key={-1}>Student</Table.HeaderCell>
              <Table.HeaderCell>Project Info</Table.HeaderCell>
              {showColumn('points') && (
                <Fragment>
                  {createHeadersTeacher() /* Week #, Code Review # */}
                  <Table.HeaderCell>Sum</Table.HeaderCell>
                </Fragment>
              )}
              {!shouldHideInstructor(props.studentInstances) && <Table.HeaderCell width="six">Instructor</Table.HeaderCell>}
              {showColumn('review') && <Table.HeaderCell>Review</Table.HeaderCell>}
            </Table.Row>
          </Table.Header>
          <Table.Body>{filteredData.map(data => createStudentTableRow(showColumn, data, rowClassName, dropDownTags, dropDownTeachers))}</Table.Body>
        </Table>
      </HorizontalScrollable>
    </Fragment>
  )
}

const mapDispatchToProps = {
  associateTeacherToStudent,
  showAssistantDropdown,
  showTagDropdown,
  selectTeacher,
  selectTag,
  filterByAssistant,
  filterByTag,
  getAllTags,
  tagStudent,
  unTagStudent,
  selectStudent,
  unselectStudent,
  selectAllStudents,
  unselectAllStudents
}

StudentTable.propTypes = {
  rowClassName: PropTypes.string,
  columns: PropTypes.array,
  allowModify: PropTypes.bool,
  disableDefaultFilter: PropTypes.bool,

  studentInstances: PropTypes.array.isRequired,
  selectedInstance: PropTypes.object.isRequired,
  coursePageLogic: PropTypes.object.isRequired,
  tags: PropTypes.object.isRequired,

  associateTeacherToStudent: PropTypes.func.isRequired,
  showAssistantDropdown: PropTypes.func.isRequired,
  showTagDropdown: PropTypes.func.isRequired,
  selectTeacher: PropTypes.func.isRequired,
  selectTag: PropTypes.func.isRequired,
  filterByAssistant: PropTypes.func.isRequired,
  filterByTag: PropTypes.func.isRequired,
  getAllTags: PropTypes.func.isRequired,
  tagStudent: PropTypes.func.isRequired,
  unTagStudent: PropTypes.func.isRequired,
  selectStudent: PropTypes.func.isRequired,
  unselectStudent: PropTypes.func.isRequired,
  selectAllStudents: PropTypes.func.isRequired,
  unselectAllStudents: PropTypes.func.isRequired
}

export default connect(
  null,
  mapDispatchToProps
)(StudentTable)
