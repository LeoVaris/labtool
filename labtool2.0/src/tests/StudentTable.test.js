import React from 'react'
import { StudentTable } from '../components/StudentTable'
import { shallow } from 'enzyme'

describe('<StudentTable />', () => {
  let wrapper

  const coursePage = {
    role: 'teacher',
    data: [
      {
        id: 10012,
        github: 'http://github.com/tiralabra2',
        projectName: 'Tiran toinen labraprojekti',
        createdAt: '2018-03-26T00:00:00.000Z',
        updatedAt: '2018-03-26T00:00:00.000Z',
        courseInstanceId: 10011,
        userId: 10012,
        teacherInstanceId: 10011,
        weeks: [],
        codeReviews: [],
        User: {
          id: 10012,
          username: 'tiraopiskelija2',
          email: 'johan.studerande@helsinki.invalid',
          firsts: 'Johan Wilhelm',
          lastname: 'Studerande',
          studentNumber: '014553242',
          admin: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z'
        },
        Tags: [
          {
            id: 20001,
            name: 'Javascript',
            color: 'red'
          }
        ]
      },
      {
        id: 10031,
        github: 'http://github.com/superprojekti',
        projectName: 'Tira super projekti',
        createdAt: '2018-03-26T00:00:00.000Z',
        updatedAt: '2018-06-05T07:12:28.603Z',
        courseInstanceId: 10011,
        userId: 10031,
        teacherInstanceId: 10011,
        weeks: [],
        codeReviews: [],
        User: {
          id: 10031,
          username: 'superopiskelija',
          email: 'teras.henkilo@helsinki.invalid',
          firsts: 'Teräs',
          lastname: 'Henkilö',
          studentNumber: '014666666',
          admin: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z'
        },
        Tags: [
          {
            id: 20008,
            name: 'DROPPED',
            color: 'grey'
          },
          {
            id: 20002,
            name: 'HTML',
            color: 'yellow'
          }
        ]
      },
      {
        id: 10011,
        github: 'http://github.com/tiralabra1',
        projectName: 'Tiran labraprojekti',
        createdAt: '2018-03-26T00:00:00.000Z',
        updatedAt: '2018-03-26T00:00:00.000Z',
        courseInstanceId: 10011,
        userId: 10011,
        teacherInstanceId: 10011,
        weeks: [],
        codeReviews: [],
        User: {
          id: 10011,
          username: 'tiraopiskelija1',
          email: 'maarit.opiskelija@helsinki.invalid',
          firsts: 'Maarit Mirja',
          lastname: 'Opiskelija',
          studentNumber: '014578343',
          admin: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z'
        },
        Tags: []
      }
    ]
  }

  const tags = {
    tags: [
      {
        id: 20001,
        name: 'Javascript',
        color: 'red',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20002,
        name: 'HTML',
        color: 'yellow',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20003,
        name: 'game',
        color: 'black',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20004,
        name: 'React',
        color: 'green',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20005,
        name: 'Node.js',
        color: 'blue',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20006,
        name: 'Java',
        color: 'orange',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20007,
        name: 'FORTRAN',
        color: 'pink',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20008,
        name: 'DROPPED',
        color: 'grey',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      }
    ]
  }

  const coursePageLogic = {
    showDropdown: '',
    selectedTeacher: '',
    filterByAssistant: 0,
    filterByTag: [],
    selectedStudents: {}
  }

  const loading = {
    loading: false,
    loadingHooks: [],
    redirect: false,
    redirectHooks: [],
    redirectFailure: false
  }

  let mockFn = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <StudentTable
        columns={['instructor']}
        studentInstances={coursePage.data}
        selectedInstance={coursePage}
        coursePageLogic={coursePageLogic}
        associateTeacherToStudent={mockFn}
        getAllTags={mockFn}
        tags={tags}
        loading={loading}
        resetLoading={mockFn}
        showAssistantDropdown={mockFn}
        showTagDropdown={mockFn}
        selectTeacher={mockFn}
        selectTag={mockFn}
        filterByAssistant={mockFn}
        filterByTag={mockFn}
        tagStudent={mockFn}
        unTagStudent={mockFn}
        selectStudent={mockFn}
        unselectStudent={mockFn}
        selectAllStudents={mockFn}
        unselectAllStudents={mockFn}
      />
    )
  })

  describe('StudentTable Component', () => {
    it('is ok', () => {
      true
    })

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
