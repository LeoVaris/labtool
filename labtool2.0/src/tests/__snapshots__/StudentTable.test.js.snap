// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StudentTable /> StudentTable Component should render correctly 1`] = `
<Fragment>
  <div
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    <span>
      Filter by instructor 
    </span>
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={true}
      closeOnEscape={true}
      deburr={false}
      defaultValue={0}
      icon="dropdown"
      minCharacters={1}
      noResultsMessage="No results found."
      onChange={[Function]}
      openOnFocus={true}
      options={Array []}
      placeholder="Select Teacher"
      renderLabel={[Function]}
      scrolling={true}
      searchInput="text"
      selectOnBlur={true}
      selectOnNavigation={true}
      selection={true}
      style={
        Object {
          "width": "NaNpx",
        }
      }
      wrapSelection={true}
    />
    <span>
       Tag filters: 
    </span>
    <span>
      <Label>
        none
      </Label>
    </span>
  </div>
  <Connect(HorizontalScrollable)>
    <Table
      as="table"
      celled={true}
      compact={true}
      singleLine={true}
      style={
        Object {
          "overflowX": "visible",
        }
      }
      unstackable={true}
    >
      <TableHeader
        as="thead"
      >
        <TableRow
          as="tr"
          cellAs="td"
        >
          <TableHeaderCell
            as="th"
            key="-1"
          >
            Student
          </TableHeaderCell>
          <TableHeaderCell
            as="th"
          >
            Project Info
          </TableHeaderCell>
          <TableHeaderCell
            as="th"
            width="six"
          >
            Instructor
          </TableHeaderCell>
        </TableRow>
      </TableHeader>
      <TableBody
        as="tbody"
      >
        <TableRow
          as="tr"
          cellAs="td"
          className="StudentRow"
          key="10012"
        >
          <TableCell
            as="td"
            key="studentinfo"
          >
            Johan Wilhelm
             
            Studerande
             (
            014553242
            )
          </TableCell>
          <TableCell
            as="td"
            key="projectinfo"
          >
            <span>
              Tiran toinen labraprojekti
              <br />
              <a
                href="http://github.com/tiralabra2"
                rel="noopener noreferrer"
                target="_blank"
              >
                http://github.com/tiralabra2
              </a>
              <div
                key="10012:20001"
              >
                <Button
                  as="button"
                  className="mini ui red button"
                  compact={true}
                  floated="left"
                  onClick={[Function]}
                >
                  Javascript
                </Button>
              </div>
            </span>
          </TableCell>
          <TableCell
            as="td"
            key="instructor"
          >
            <span>
              not assigned
            </span>
          </TableCell>
        </TableRow>
        <TableRow
          as="tr"
          cellAs="td"
          className="StudentRow"
          key="10031"
        >
          <TableCell
            as="td"
            key="studentinfo"
          >
            Teräs
             
            Henkilö
             (
            014666666
            )
          </TableCell>
          <TableCell
            as="td"
            key="projectinfo"
          >
            <span>
              Tira super projekti
              <br />
              <a
                href="http://github.com/superprojekti"
                rel="noopener noreferrer"
                target="_blank"
              >
                http://github.com/superprojekti
              </a>
              <div
                key="10031:20008"
              >
                <Button
                  as="button"
                  className="mini ui grey button"
                  compact={true}
                  floated="left"
                  onClick={[Function]}
                >
                  DROPPED
                </Button>
              </div>
              <div
                key="10031:20002"
              >
                <Button
                  as="button"
                  className="mini ui yellow button"
                  compact={true}
                  floated="left"
                  onClick={[Function]}
                >
                  HTML
                </Button>
              </div>
            </span>
          </TableCell>
          <TableCell
            as="td"
            key="instructor"
          >
            <span>
              not assigned
            </span>
          </TableCell>
        </TableRow>
        <TableRow
          as="tr"
          cellAs="td"
          className="StudentRow"
          key="10011"
        >
          <TableCell
            as="td"
            key="studentinfo"
          >
            Maarit Mirja
             
            Opiskelija
             (
            014578343
            )
          </TableCell>
          <TableCell
            as="td"
            key="projectinfo"
          >
            <span>
              Tiran labraprojekti
              <br />
              <a
                href="http://github.com/tiralabra1"
                rel="noopener noreferrer"
                target="_blank"
              >
                http://github.com/tiralabra1
              </a>
            </span>
          </TableCell>
          <TableCell
            as="td"
            key="instructor"
          >
            <span>
              not assigned
            </span>
          </TableCell>
        </TableRow>
      </TableBody>
    </Table>
  </Connect(HorizontalScrollable)>
</Fragment>
`;
